snowflake/datadog_checks/snowflake/check.py

```diff
@@ -116,6 +121,9 @@ def connect(self):
             self.proxy_port,
         )
 
+        if self._config.token_path:
```

Comment:
In [vault](https://github.com/DataDog/integrations-core/blob/master/vault/datadog_checks/vault/vault.py#L124), we only refresh the token if there's an error, what do you think about that approach here?

Reply:
It is possible, but since snowflake runs once every hour by default, I think it's better to read the token at each run
We try one less connection, we're sure to use the latest token, and the code and logs are more readable

Reply:
Ah true, that makes sense with snowflake's collection interval.